pkg/nats/stream/consumer_client_test.go

```diff
@@ -71,7 +71,7 @@ func (suite *ClientTestSuite) TestRequestWithContextCancellation() {
 	cancel()
 
 	// Attempt to make the request
-	_, err := suite.streamingClient.OpenStream(ctx, subj, payload)
+	_, err := suite.streamingClient.OpenStream(ctx, subj, "", payload)
```

Comment:
I completely understand your concern for blank producer identifiers. Ideally it should be fixed. 

However, I don't think `subject` would make any things better. 

Lets take an example. Say we have a `Consumer 1` and `Producer 1`. Tomorrow may be `Consumer 1` and `Producer 1` are communicating on multiple request subjects like `node-Producer1.ExecutionLogRequest/V1` and `node-Producer2.ContainerNetworkLogs`. 

If `Producer 1` starts collecting stream ids in grouping them based on request subjects, there is no value which consumer gets from it. With heart beating the idea is to determine if the connection between the producer and connection client is alive. To determine liveness, we should just be worried about the connection details. 

Let me know if you think otherwise ?